### DISCUSSION TITLE
Nested / Group Data in Table View

### DISCUSSION THREAD

**Discussion by Leo831-ux** [Budibase/budibase#14530] [Help] [answered]
Hi

I am in the process of porting my app from Nocodb to Budibase when I got stuck trying to replicate the view below in Budibase.

![Screenshot 2024-09-06 095412](https://github.com/user-attachments/assets/7e6265c9-0b3a-400f-8f97-af50e2c00246)

May, I kindly ask for help or pointers on how to accomplish the view. This view will be available to the client.

**Comment by mikesealey** [comment]
Hey @Leo831-ux Could you provide a little more context? Is the screenshot what you'd like to replicate using Budibase? 

Are Sites the rows in a table that you'd like to iterate over? What does your data structure look like?

**Comment by Leo831-ux** [comment]
Hi

I am trying to replicate the view in the screenshot within Budibase

I have also attached the app below.  Data that would be used is from the Cash Drop table

In essence I would like to group the data according to the site, then date and then the entry as it appears in the Cash Drop table.
I hope it make sense.

[CLV Group-export-1725636051695.tar.gz](https://github.com/user-attachments/files/16909974/CLV.Group-export-1725636051695.tar.gz)

**Reply by mikesealey** [reply, marked as answer]
A few notes:

Looking at your Cash Drop table you have "Date" which is a date column, but you then have a text column - "Time" but you can capture Date and Time in a Date column by making sure that "Date Only" is unchecked.

You also have Sites, Staff, View, Reason, all as Text columns, but it looks like you might be happier having them as relationships. You could also have a View set as a Single Select column, that would look a little bit like this: 
![image](https://github.com/user-attachments/assets/dc683c4d-db35-45d5-a9c7-75f0320f07ea)

I've made a little mock-up based on what you provided, but it's far from a finished product. We offer Proof-of-Concept App building services to our Enterprise customers, so if that's something you'd like to consider please contact our Sales team.

So, starting at the bottom of the component tree, I've added a Data Provider "PARENT Data Provider" which fetches all of the data from the Cash Drop table. 

Then, I've added a container that holds a Data Provider that fetches data from the PARENT data provider, but filters over the Site column, to make sure that it's only getting Site A data. Inside that is a Repeater component that will show a copy of it's children for every row it finds. There are extra little cosmetics, but roughly what you're looking for is there. Last piece of the puzzle is a sibling of the Repeater, simply showing how many rows that the Data Provider has found matching those filters.

![image](https://github.com/user-attachments/assets/865d4531-eed0-4613-92b7-dc622d7b4d9c)

Here's an export for you to take a closer look - I added my work on a screen called /mike1 - you might be happiest creating a new app to import this, so as not to overwrite any changes you've made since you exported. My work here is only for demonstration purposes
[14530 Export-export-1725651108535.tar.gz](https://github.com/user-attachments/files/16912261/14530.Export-export-1725651108535.tar.gz)

The reason the Data Providers have been set up in this way is to minimise the number of calls to the database. the PARENT data provider will fetch all of the information once, and then the data providers that are set to the Parent as a source will not need to make additional requests, as the data is all already fetched, the just filter it locally. At this level of the app it won't make a huge difference, but as the size and complexity of your app grows it will greatly improve performane.

**Comment by Leo831-ux** [comment]
Hi Mike

Thank you very much. I appreciate all the tips and tricks, since this is my very first app with Budibase.

I will be spending the next couple of days going through your screen.  It is solid foundation and an excellent platform for me to learn from, in addition to getting an idea of how things fit together. 

Have you guys ever thought of maybe putting together a paid advance course that covers more advance scenarios and designs?

I was thinking of having a single select for the View columns but decided against it.  There is a good possibility that the information in the table will be used with another table and view.  In addition, I needed to give the employees in office access to adding new and edit entries without direct access to the database.  The screen / form for that functionality still needs to be added.
